satpy/readers/hdf4_utils.py

```diff
@@ -45,12 +47,24 @@
 }
 
 
-def from_sds(var, *args, **kwargs):
+def from_sds(var, src_path, *args, **kwargs):
     """Create a dask array from a SD dataset."""
-    var.__dict__["dtype"] = np.dtype(HTYPE_TO_DTYPE[var.info()[3]])
-    shape = var.info()[2]
+    var_info = var.info()
+    var.__dict__["dtype"] = np.dtype(HTYPE_TO_DTYPE[var_info[3]])
+    shape = var_info[2]
     var.__dict__["shape"] = shape if isinstance(shape, (tuple, list)) else tuple(shape)
-    return da.from_array(var, *args, **kwargs)
+
+    name = kwargs.pop("name", None)
```

Comment:
I'm wondering if the filename should be included in the name also. I'm thinking of the case where the user opens two 1KM modis file at the same time. The item names will be the same...

Reply:
So essentially, we could append the token generated after to the existing name if not None.

Reply:
Yeah, it is important to be able to open multiple files at once so appending the filename sounds like a good way to ensure this.

Reply:
I'm sorry, I think you're misunderstanding. The `src_path` is included in the token already on line 60:

```
tokenize_args = (os.fspath(src_path), var_name)
```

It is passed to `tokenize`. This `name = kwargs.pop` line is just in case someone wants to force the dask name to something else. In that case, my new tokenize logic is skipped over.

Reply:
Ok, sorry! In that case it means we don't have any test for that part of the code I think since codecov is complaining.

Reply:
No, codecov is complaining specifically for line 62 because we don't have a realworld case where `*args` are ever passed to this function. I could remove `*args` since it is unlikely that we'd provide them since I don't think dask `from_array` takes any positional arguments.

Reply:
Ok I've done this. It was never used and I was right `from_array` doesn't accept any other positional arguments besides the array-like first argument.

Reply:
👌, thanks a lot